module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,47 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <forward_list>
+#include <iostream>
+#include <iterator>
+
+compressedImage compressGrayscale(const image& bitmap) {
+    compressedImage compression;
+    compression.reserve(width * height);
+
+    for (const auto& row : bitmap) {
+        for (auto it = row.begin(); it != row.end();) {
+            auto newIt = std::adjacent_find(it, row.end(), [](auto a, auto b) { return a != b; });
```

Comment:
In `<funcional>` there is prepared function `not_equal_to<T>` or smth like that :) I always good to use existing functions, because bilions of people used it before ;p Maybe not billions, but more than one ;)

Reply:
oh ye, you are right. good to know!